File: bench/python/regex.py
# Generate N decimal strings; count full matches of [0-9]+. Checksum = count.
import os, re
N = int(os.environ.get("BENCH_N", "20000"))
pat = re.compile(r"[0-9]+")
x = 123456789
count = 0
for _ in range(N):
    x = (x * 1103515245 + 12345) & 0x7FFFFFFF
    s = str(x)
    if x % 2 == 0:
        s += "x"
    if pat.fullmatch(s):
        count += 1
print(count)

File: bench/clojure/regex.clj
;; Generate N decimal strings; count full matches of [0-9]+ (re-matches is
;; anchored). Checksum = count.
(let [n (long (Long/parseLong (or (System/getenv "BENCH_N") "20000")))
      re #"[0-9]+"]
  (loop [i 0 x 123456789 count 0]
    (if (< i n)
      (let [x (bit-and (+ (* x 1103515245) 12345) 0x7FFFFFFF)
            s0 (str x)
            s (if (even? x) (str s0 "x") s0)]
        (recur (inc i) x (if (re-matches re s) (inc count) count)))
      (println count))))
